### PR TITLE
[WS-pscan-scripts_js] Add Error Application Disclosure Script

### PR DESCRIPTION
**Pull request by eakirtas**
I split the Error Application Disclosure scanner into two scripts one is using string and the other using regex.

**NOTE:** Application error strings are equal to (the only difference is that characters '/' is escaped -> '//' )

_______________________________________________________________________
**Test Example:** 
```JS
json-test : {
	  "base64-no-=" : "SGVsbG8gV29ybGQh",
	  "base64-with-=" : "SGVsbG8gV29ybGQ=",
	  "non-base64" : "Hello World",
	  "non-base64-with" : "1+2=3",
	  "non_base64_with_2" : "1234==",
	  "an-email": "example@email.com",
    "application-error": "Warning: Cannot modify header information - headers already sent",
    "app-error-regex_1" : "Line 54: Incorrect syntax near 'here'"
    "app-error-regex_2" : "pg_query(): query failed: error: ",
    "app-error-regex_3" : "pg_query(): Query failed: ERROR: column \"tom\" does not exist LINE 2: SET name=tom",
    "app-error-regex_4" : "'it's a text' is null or not an object",
    "app-error-regex_5" : "ORA-30625: method dispatch on NULL SELF argument is disallowed\n"
        + "Cause: A member method of a type is being invoked with a NULL SELF argument.\n"
        + "Action: Change the method invocation to pass in a valid self argument.",
    "app-error-regex_6 : "System.Data.OleDb.OleDbException: Syntax error in string in query expression 'User ID = ? And Password = ?'"
}

}
```

### CODE REVIEW

File: addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/application_error_scanner.js [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+        if((matches = message.match(pattern)) != null){
+            matches.forEach(function(evidence){
+
+                helper.newAlert()
+                    .setRiskConfidence(RISK_MEDIUM, CONFIDENCE_MEDIUM)
+                    .setName("Application Error Disclosure via WebSockets (script)")
+                    .setDescription(
+                        "This page contains an error/warning message that \
+ may disclose sensitive information like the location of the file \
+ that produced the unhand-led exception. This information can be used\
```
Comment (kingthorin) [inline addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/application_error_scanner.js]: "unhand-led" > "unhandled"

File: addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/application_error_scanner.js [hunk 2]
```diff
@@ -0,0 +1,137 @@
+// * This Script analyze incoming websocket messages for error messages
```
Comment (kingthorin) [inline addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/application_error_scanner.js]: "analyze" > "analyzes"

File: addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/application_error_scanner_regex.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
+// * This Script analyze incoming websocket messages for error messages
```
Comment (kingthorin) [inline addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/application_error_scanner_regex.js]: "analyze" > "analyzes"

File: addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/application_error_scanner_regex.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+// * Based on org.zaproxy.zap.extension.pscanrules.ApplicationErrorScanner
+// * Application error strings are equal to (characters '/' is escaped -> '//'):
+// ** https://github.com/zaproxy/zap-extensions/blob/master/addOns/pscanrules/src/main/resources/org/zaproxy/zap/extension/pscanrules/resources/application_errors.xml
```
Comment (kingthorin) [inline addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/application_error_scanner_regex.js:5]: Perhaps this one should have a comment that differentiates it from the one above?

File: addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/application_error_scanner.js [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+ may disclose sensitive information like the location of the file \
+ that produced the unhand-led exception. This information can be used\
+ to launch further attacks against the web application. The alert\
+ could be a false positive if the error message is found inside a\
+ documentation page.")
+                    .setSolution("Review the source code of this page. Implement custom \
+ error pages. Consider implementing a mechanism to provide a unique\
+ error reference/identifier to the client (browser) while logging the\
+ details on the server side and not exposing them to the user.")
+                    .setParam(counter++)
```
Comment (kingthorin) [inline addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/application_error_scanner.js]: I think this is probably used on some handling in the [Find HTML Comments](https://github.com/zaproxy/community-scripts/blob/27efdf1cb39dee4f9dd71d7055e4b5a6ee231c42/passive/Find%20HTML%20Comments.js#L23-L24) script. However: 1) It's an older script. 2) It uses parameters in order to facilitate different grouping of results.

In the majority of cases passive scanners (scripts or other) should not really have/need a "parameter".

File: addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/application_error_scanner_regex.js [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+ may disclose sensitive information like the location of the file \
+ that produced the unhandled exception. This information can be used\
+ to launch further attacks against the web application. The alert\
+ could be a false positive if the error message is found inside a\
+ documentation page.")
+                .setSolution("Review the source code of this page. Implement custom \
+ error pages. Consider implementing a mechanism to provide a unique\
+ error reference/identifier to the client (browser) while logging the\
+ details on the server side and not exposing them to the user.")
+                .setParam(counter++)
```
Comment (kingthorin) [inline addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/application_error_scanner_regex.js]: I think this is probably used on some handling in the [Find HTML Comments](https://github.com/zaproxy/community-scripts/blob/27efdf1cb39dee4f9dd71d7055e4b5a6ee231c42/passive/Find%20HTML%20Comments.js#L23-L24) script. However: 1) It's an older script. 2) It uses parameters in order to facilitate different grouping of results.

In the majority of cases passive scanners (scripts or other) should not really have/need a "parameter".

File: addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/application_error_scanner.js (hunk 1 above)
Comment (eakirtas) [reply addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/application_error_scanner.js]: Fixed in cd5c145

File: addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/application_error_scanner.js (hunk 2 above)
Comment (eakirtas) [reply addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/application_error_scanner.js]: Fixed in cd5c145

File: addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/application_error_scanner_regex.js (hunk 3 above)
Comment (eakirtas) [reply addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/application_error_scanner_regex.js]: Fixed in cd5c145

File: addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/application_error_scanner_regex.js (hunk 4 above)
Comment (eakirtas) [reply addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/application_error_scanner_regex.js:5]: Fixed in cd5c145

File: addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/application_error_scanner.js (hunk 5 above)
Comment (eakirtas) [reply addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/application_error_scanner.js]: Fixed in 319f60c

File: addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/application_error_scanner_regex.js (hunk 6 above)
Comment (eakirtas) [reply addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/application_error_scanner_regex.js]: Fixed in 319f60c

**Pull request closed without merging**